tests/h/services/profile_group_test.py

```diff
@@ -11,43 +11,106 @@
-        svc = ProfileGroupService(db_session, open_group_finder=auth_group_svc.public_groups)
+    def test_all_returns_open_groups_for_no_user(self, profile_group_factory, open_groups):
+        svc = profile_group_factory(open_groups)
 
         groups = svc.all(user=None)
 
         assert [group['id'] for group in groups] == [group.pubid for group in open_groups]
 
-    def test_all_proxies_authority_parameter_to_svc(self, db_session):
+    def test_sort_sorts_private_groups_by_name(self, profile_group_factory, factories):
+        user = factories.User()
+        user.groups = [factories.Group(name='zedd'),
+                       factories.Group(name='alpha')]
+        svc = profile_group_factory([])
+
+        groups = svc.all(user)
+
+        names = [group['name'] for group in groups]
+        assert names == ['alpha', 'zedd']
+
+    def test_sort_sorts_private_groups_by_pubid(self, profile_group_factory, factories):
+        user = factories.User()
+        user.groups = [factories.Group(name='zedd', pubid='yyyyy'),
+                       factories.Group(name='alpha', pubid='zzzzz'),
+                       factories.Group(name='zedd', pubid='aaaaa')]
+        svc = profile_group_factory([])
+
+        groups = svc.all(user)
+
+        sorted_groups = [(group['id'], group['name']) for group in groups]
+        assert sorted_groups == [('zzzzz', 'alpha'), ('aaaaa', 'zedd'), ('yyyyy', 'zedd')]
+
+    def test_sort_sorts_public_groups_by_name(self, profile_group_factory, factories):
+        svc = profile_group_factory([
+            factories.OpenGroup(name='zedd', pubid='aaaaa'),
+            factories.OpenGroup(name='alpha', pubid='zzzzz')
+        ])
+
+        groups = svc.all()
+
+        names = [group['name'] for group in groups]
+        assert names == ['alpha', 'zedd']
+
+    def test_sort_sorts_public_groups_by_pubid(self, profile_group_factory, factories):
+        svc = profile_group_factory([
+            factories.OpenGroup(name='zedd', pubid='yyyyy'),
+            factories.OpenGroup(name='zedd', pubid='aaaaa'),
+            factories.OpenGroup(name='alpha', pubid='zzzzz')
+        ])
+
+        groups = svc.all()
+
+        sorted_groups = [(group['id'], group['name']) for group in groups]
+        assert sorted_groups == [('zzzzz', 'alpha'), ('aaaaa', 'zedd'), ('yyyyy', 'zedd')]
+
+    def test_sort_sorts_groups_by_type(self, profile_group_factory, factories):
+        """ open groups should come first """
+        user = factories.User()
+        user.groups = [
+            factories.Group(name='zebra2', pubid='apple'),
+            factories.Group(name='zebra', pubid='zebra')
+        ]
+        svc = profile_group_factory([
+            factories.OpenGroup(name='zedd', pubid='yyyyy'),
+            factories.OpenGroup(name='alpha', pubid='zzzzz')
+        ])
+
+        groups = svc.all(user)
+
+        names = [group['name'] for group in groups]
+        assert names == ['alpha', 'zedd', 'zebra', 'zebra2']
```

Comment:
These sort tests are great as they are but they look like they might be amenable to being parametrized, removing some duplication.

Reply:
I'll take a peek! It's my recollection that the values were changing enough (by necessity) between tests to make this impracticable, but I'll check.